src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -72,22 +72,51 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
-        }
+        // Find the person to edit based on name or ID
+        Optional<Person> personOptional = findPersonToEdit(lastShownList);
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
+        Person personToEdit = personOptional.get();
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
-        }
-
```

Comment:
I have concerns over removing this check. Do we currently have an alternative check to editing in a duplicate person?

Reply:
This is addressed in the latest commit.